packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4055,6 +4088,34 @@ class InheritedElement extends ProxyElement {
     super.debugDeactivated();
   }
 
+  /// Returns the most recent [updateDependencies] value for [dependent].
+  ///
+  /// Each dependent element is mapped to a single [InheritedDependencies]
+  /// object.
+  @protected
+  InheritedDependencies getDependencies(Element dependent) => _dependents[dependent];
+
+  /// Called by [inheritFromWidgetOfExactType] when a new [dependent] is added.
+  ///
+  /// Each dependent element is mapped to a single [InheritedDependencies]
+  /// object. This method can lookup the existing dependencies with
+  /// [getDependencies].
+  ///
+  /// By default this method returns a constant [InheritedDependencies] object
+  /// for which [InheritedDependencies.isGlobal] is true.
+  @protected
+  InheritedDependencies updateDependencies(Element dependent, Object aspect) {
```

Comment:
I'm a bit confused by this method being called `update` but returning a value. It took me a while to understand the model here. I'm not sure what to suggest, though...

Maybe if it took the current dependency as an argument?

Reply:
I could have explained these methods better, sorry about that. 

If there isn't a dependency (yet) then the "current dependency" is null: `getDependencies(dependent)` returns null. This method is responsible for constructing a a new dependency or updating an existing one. 

I guess this method's signature is more like the Map.update method than Element.update.

Reply:
Map.update actually changes the map, though, which this method doesn't.

Maybe it would be less confusing if the InheritedDependencies object was immutable... though that would be a lot more expensive since you'd be copying a Set all the time...

> If there isn't a dependency (yet) then the "current dependency" is null

I think this is part of what is confusing. When it's null, this method is pure-functional, it just returns a new value, and the return value is critical. When it's not null, it's destructive, it actually changes the underlying value, and if the return value was ignored it would still work.

Reply:
I added a setDependencies() method which can be used by updateDependencies() :

```
  InheritedDependencies getDependencies(Element dependent) {
    return _dependents[dependent];
  }

  void setDependencies(Element dependent, InheritedDependencies value) {
    _dependents[dependent] = value;
  }

  void updateDependencies(Element dependent, Object aspect) {
    setDependencies(dependent, const InheritedDependencies()); // isGlobal => true
  }
```